docs/faq.md

```diff
@@ -366,3 +366,9 @@ If you can avoid using these features, you can avoid triggering the error. The o
    Excluding mutation webhooks from the diff could cause undesired diffing behavior.
 3. **Disable mutation webhooks when using server-side diff**: see [server-side diff docs](user-guide/diff-strategies.md#mutation-webhooks)
    for details about that feature. Disabling mutation webhooks may have undesired effects on sync behavior.
+
+
+## Why does Argo CD not detect changes in deployment's environment variables ?
```

Comment:
Should we make this more general? It's not just env variables that get impacted, it's also anything not declared in GIT. Eg, https://github.com/argoproj/argo-cd/issues/14838

Reply:
So basically any changes after the creation except the images and replicas are ignored ?

What I'm still not sure is if there's some difference in diff detection if you add something by declaring it in git vs not declared in git. In the case of additions of environment variables, they are ignored even though they are declared in git.